corehq/motech/repeaters/models.py

```diff
@@ -904,11 +904,11 @@ def already_processed():
         if already_processed() or not is_ready():
             return
 
-        # Set the next check to happen an arbitrarily long time from now so
-        # if something goes horribly wrong with the delayed task it will not
-        # be lost forever. A check at this time is expected to occur rarely,
-        # if ever, because `process_repeat_record` will usually succeed or
-        # reset the next check to sometime sooner.
+        # Set the next check to happen an arbitrarily long time from now.
+        # This way if there's a delay in calling `process_repeat_record` (which
+        # also sets or clears next_check) we won't queue this up in duplicate.
+        # If `process_repeat_record` is totally borked, this future date is a
+        # fallback.
```

Comment:
Incorporating review comment into the code comment ...

```suggestion
        # If `process_repeat_record` is totally borked, and errors out before
        # `next_check` is assigned a value, then the repeat record will never be
        # sent, and effectively lost forever. Setting a value here allows a
        # fallback.
```